sklearn/neighbors/lmnn.py

```diff
@@ -0,0 +1,1359 @@
+# coding: utf-8
+"""
+Large Margin Nearest Neighbor Classification
+"""
+
+# Author: John Chiotellis <johnyc.code@gmail.com>
+# License: BSD 3 clause
+
+from __future__ import print_function
+from warnings import warn
+
+import sys
+import time
+import numpy as np
+from scipy.optimize import minimize
+from scipy.sparse import csr_matrix, csc_matrix, coo_matrix
+
+from ..base import BaseEstimator, TransformerMixin
+from ..pipeline import Pipeline
+from ..neighbors import NearestNeighbors, KNeighborsClassifier
+from ..decomposition import PCA
+from ..utils import gen_batches
+from ..utils.extmath import row_norms, safe_sparse_dot
+from ..utils.random import check_random_state
+from ..utils.multiclass import check_classification_targets
+from ..utils.validation import check_is_fitted, check_array, check_X_y
+from ..externals.six import integer_types, string_types
+
+
+class LargeMarginNearestNeighbor(BaseEstimator, TransformerMixin):
+    """Distance metric learning for large margin classification.
+
```

Comment:
Please add a "Read more in the User Guide" like other estimator docstrings

Reply:
Done. Thanks @jnothman and happy new year!